lib/auth/trustedcluster.go

```diff
@@ -221,13 +230,16 @@ func (a *Server) DeleteTrustedCluster(ctx context.Context, name string) error {
 		return trace.BadParameter("trusted cluster %q is the name of this root cluster and cannot be removed.", name)
 	}
 
-	// Remove all CAs
-	for _, caType := range []types.CertAuthType{types.HostCA, types.UserCA, types.DatabaseCA, types.OpenSSHCA} {
-		if err := a.DeleteCertAuthority(ctx, types.CertAuthID{Type: caType, DomainName: name}); err != nil {
-			if !trace.IsNotFound(err) {
-				return trace.Wrap(err)
-			}
-		}
+	var ids []types.CertAuthID
+	for _, caType := range trustedClusterCATypes {
```

Comment:
For enhanced paranoia, should we delete all the known types (in `types.CertAuthTypes`)?